motoman_msgs/srv/SelectTool.srv

```diff
@@ -0,0 +1,92 @@
+
+# Changes the tool file used by the controller for (Moto)ROS-controlled motions.
+#
+# NOTE 1: this changes the active tool for (Moto)ROS-controlled motions ONLY.
+#
+#         Regular INFORM programs, jogging or any other (programmed) motions
+#         executed by the controller will NOT use the new tool.
+#
+# NOTE 2: any tool switch initiated by invoking this service WILL however be
+#         taken into account by safety systems such as the FSU and PFL while
+#         executing (Moto)ROS-initiated motions.
+#
+# NOTE 3: active tool will be switched AFTER the controller has completed
+#         execution of the last trajectory segment that was sent to the
+#         controller BEFORE this service was called.
+#
+#         The motion queue internal to MotoROS caches a number of segments in
```

Comment:
@ted-miller @EricMarcil: is there any way we can make this more exact (as in: number of seconds of motion, number of samples, something else)?

Reply:
Not necessarily. It's not just the max length of the incremental queue. There could also be pending data that hasn't been broken down yet because the queue is full.